packages/fast-components-react-base/src/dialog/dialog.tsx

```diff
@@ -231,18 +233,28 @@ class Dialog extends Foundation<DialogHandledProps, DialogUnhandledProps, {}> {
      */
     private handleDocumentFocus = (event: FocusEvent): void => {
         if (
-            this.props.visible &&
             !event.defaultPrevented &&
-            this.rootElement.current instanceof HTMLElement &&
-            !(this.rootElement.current as HTMLElement).contains(
-                event.target as HTMLElement
-            )
+            this.shouldForceFocus(event.target as HTMLElement)
         ) {
             this.focusOnFirstElement();
             event.preventDefault();
         }
     };
 
+    /**
+     * test to avoid forcing focus when focus is already within
+     */
+    private shouldForceFocus = (currentFocusElement: Element): boolean => {
+        if (
+            this.props.visible &&
```

Comment:
This conditional evaluates to true/false, so you can reduce this to just return the product of the conditional:

```
return this.props.visible && this.rootElement.current instanceof HTMLElement && !(this.rootElement.current as HTMLElement).contains(currentFocusElement)